ci, doc: Add a link to Android APK CI artifact

hebasto opened the pull request:
Ability to download Android APK from CI was requested on the recent [call](https://github.com/BitcoinDesign/Meta/issues/185).

[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/82)
[![macOS](https://img.shields.io/badge/OS-macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/82)
[![Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/82)

hebasto commented:
@promag Could you suggest a better Android badge link?

promag reviewed (commented):
Concept ACK.

Maybe change to:
https://img.shields.io/badge/OS-Windows-green
https://img.shields.io/badge/OS-macOS-green
https://img.shields.io/badge/OS-Android-green

hebasto commented:
> Concept ACK.
> 
> Maybe change to: https://img.shields.io/badge/OS-Windows-green https://img.shields.io/badge/OS-macOS-green https://img.shields.io/badge/OS-Android-green

Thank you @promag! Done.

hebasto commented:
@Bosch-0 

Is this what you were requested for?

promag commented on .cirrus.yml:
```diff
@@ -320,3 +321,7 @@ task:
+  copy_artifacts_script:
+    - cp src/qt/android/build/outputs/apk/release/android-release-unsigned.apk ${CIRRUS_WORKING_DIR}/insecure_android.apk
+  insecure_android_apk_artifacts:
+    path: "insecure_android.apk"
```
nit, add newline.

hebasto replied:
Done.

promag reviewed (commented):
It failed to install, but link works.

jarolrod reviewed (commented):
Concept ACK

Can't test the resulting apk, but the link works

hebasto commented:
> It failed to install, but link works.

Just leaving here a [link](https://github.com/bitcoin/bitcoin/pull/22469#issuecomment-881867614) to @icota's comment dated 2021-07-17.

icota commented:
```
adb: failed to install insecure_android.apk: Failure [INSTALL_PARSE_FAILED_NO_CERTIFICATES: Failed collecting certificates for /data/app/vmdl895240733.tmp/base.apk: Package /data/app/vmdl895240733.tmp/base.apk has no certificates at entry AndroidManifest.xml]
```
Not sure why it worked before. Maybe `gradle` somehow found my dev cert from outside the container.

hebasto commented:
@promag
> It failed to install, but link works.

Mind trying one more time?

promag commented:
@hebasto same error.

promag commented:
Tested ACK e9a1e98fa44547afcc9d284c38dc0442159abf7f

![image](https://user-images.githubusercontent.com/3534524/141338852-7b2f14c8-87e1-4e39-a169-a57934d758d1.png)

hebasto commented:
Updated e9a1e98fa44547afcc9d284c38dc0442159abf7f -> 83f1893265bfa9b45dfc08cd2cf6d91c451e213d ([pr82.03](https://github.com/hebasto/gui-qml/commits/pr82.03) -> [pr82.04](https://github.com/hebasto/gui-qml/commits/pr82.04), [diff](https://github.com/hebasto/gui-qml/compare/pr82.03..pr82.04)):

- fixed filename in README.md

promag reviewed (commented):
ACK 83f1893265bfa9b45dfc08cd2cf6d91c451e213d, just fixed url in readme.

shaavan reviewed (commented):
Concept ACK

I was able to download the android apk through the link successfully:
```
https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/82
```

The app is working perfectly well on my phone (mi note 5 pro).

**Screenshot:**

![WhatsApp Image 2021-11-12 at 4 23 11 PM](https://user-images.githubusercontent.com/85434418/141457582-01e0340c-4c90-4227-b481-6351aca9176e.jpeg)

However, when I clicked on the link in the readme file:
```
https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip
```

I received the following error page message:

![Screenshot from 2021-11-12 12-56-52](https://user-images.githubusercontent.com/85434418/141457634-5025b88d-1130-4321-9c6d-123caf3ddd6d.png)

I am not sure, but this seems like an unexpected thing and should be taken a look at.

**On a side note:**

I noticed a thing while installing the app on my phone. While installing, the shown icon of the app was a default package icon.

![WhatsApp Image 2021-11-12 at 4 35 06 PM](https://user-images.githubusercontent.com/85434418/141457680-4da2701e-d9f6-4ed6-a047-a7971e23f22c.jpeg)

But after installing the bitcoin core icon showed:

![WhatsApp Image 2021-11-12 at 4 35 06 PM(1)](https://user-images.githubusercontent.com/85434418/141457721-1a4546c9-d3a1-42d2-a6e4-3ead4c34b0ab.jpeg)

This bug could be taken up in a follow-up PR!

promag commented:
@shaavan I think it will only work when this gets merged.

shaavan reviewed (approved):
> I think it will only work when this gets merged.

If that's the case, then I find no issue with this PR :)

ACK 83f1893265bfa9b45dfc08cd2cf6d91c451e213d

shaavan commented:
> I think it will only work when this gets merged.

@promag @hebasto, the link (https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip) is still not working.

promag commented:
@shaavan lets wait master jobs finishes.

promag commented:
@shaavan it works
